Attempting to `cargo install --path .` for the Summation.hvm example from the HVM guide readme fails

I followed the installation instructions in HVM/guide/README.md, and created the summation.hvm file.

In the Compiling A Program section, when trying to run `cargo install --path .` I get the following errors:

```
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> src/lib.rs:1:12
  |
1 | #![feature(atomic_from_mut)]
  |            ^^^^^^^^^^^^^^^

error[E0554]: `#![feature]` may not be used on the stable release channel
 --> src/lib.rs:2:12
  |
2 | #![feature(atomic_mut_ptr)]
  |            ^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0554`.
```

I'm currently running `hvm 1.0.13-beta`, and *did* use the nightly options when setting up rust and cargo.
My OS: MacOS 12.6.1.

--- a/flake.nix
+++ b/flake.nix
@@ -7,12 +7,15 @@
     };
     nci = {
       inputs.nixpkgs.follows = "nixpkgs";
-      # The next commit, 5d3d4b15b7a5f2f393fe60fcdc32deeaab88d704, is broken.
-      url = "github:yusdacra/nix-cargo-integration/774b49912e6ae219e20bbb39258f8a283f6a251c";
+      url = "github:yusdacra/nix-cargo-integration";
     };
+    nix-filter.url = "github:numtide/nix-filter";
     nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
   };
-  outputs = inputs:
+  outputs = inputs: let
+    nix-filter = import inputs.nix-filter;
+    pkgs = common: packages: builtins.map (element: common.pkgs.${element}) packages;
+  in
     inputs.nci.lib.makeOutputs {
       config = common: {
         cCompiler = {
@@ -25,11 +28,11 @@
             package = "hvm";
           };
         };
-        runtimeLibs = [common.pkgs.openssl];
+        runtimeLibs = pkgs common ["openssl"];
         shell = {commands = builtins.map (element: {package = common.pkgs.${element};}) ["ghc" "nodejs"];};
       };
       pkgConfig = common: let
-        override = {buildInputs = [common.pkgs.openssl common.pkgs.pkg-config];};
+        override = {buildInputs = pkgs common ["openssl" "pkg-config"];};
       in {
         hvm = {
           app = true;
@@ -39,11 +42,19 @@
           profiles = {
             dev = false;
             dev_fast = false;
-            release = true;
+            release = false;
           };
         };
       };
-      # Exclude "wasm" directory because it causes "error: attribute 'crane' missing" and "error: expected a derivation".
-      root = builtins.filterSource (path: type: !(type == "directory" && baseNameOf path == "wasm")) ./.;
+      # Only include directories necessary for building the project, to make the derivation smaller.
+      root = nix-filter {
+        root = ./.;
+        include = [
+          ./src
+          ./Cargo.lock
+          ./Cargo.toml
+          ./rust-toolchain.toml
+        ];
+      };
     };
 }

--- a/src/compiler/mod.rs
+++ b/src/compiler/mod.rs
@@ -12,6 +12,7 @@ pub fn compile(code: &str, name: &str) -> std::io::Result<()> {
   // hvm
   std::fs::create_dir(format!("./{}",name)).ok();
   std::fs::write(format!("./{}/Cargo.toml",name), cargo_rs)?;
+  std::fs::write(format!("./{}/rust-toolchain.toml",name), include_str!("./../../rust-toolchain.toml"))?;
 
   // hvm/src
   std::fs::create_dir(format!("./{}/src",name)).ok();

